Phase 1

meech-ward opened the pull request.

meech-ward committed:
Remove Essential Chapters from BNR intro

meech-ward committed:
Add a Getting Started with Objective-C section

meech-ward committed:
Add Primitive Data Types challenge

meech-ward committed:
Add Intro to Objects and NSString challenge

meech-ward committed:
Add More Objects challenge

meech-ward committed:
Add Collections challenge

meech-ward committed:
Add Creating Custom Classes challenge

meech-ward committed:
Add Custom Methods challenge

meech-ward committed:
Add Custom Methods challenge

meech-ward committed:
Add Properties challenge

meech-ward committed:
Add Initializers challenge

meech-ward committed:
Add Inheritance challenge

meech-ward committed:
Add More Resources section

meech-ward committed:
Update the order of the assignments

meech-ward committed:
Add the remaining lynda videos to Nice to Have

Sections 9, 10, and 11 were not added to any of the challenges so students should do them in the nice to have sections.

meech-ward committed:
Fix spelling mistakes

meech-ward committed:
Add a disclaimer that bnr chapters can be different

The chapter numbers in Big Nerd Ranch can vary between different edditions of the book.

meech-ward committed:
Remove any links to Big Nerd Ranch

Remove all the link to the big nerd ranch chapters. Most of the links were to safari books online, but if the stude nt doesn't have a safari account, then the link doesn't really do much for them.

review-ninja commented:
<a href="https://app.review.ninja/lighthouse-labs/ios-prep-book/pull/38" target="_blank"><img src="https://app.review.ninja/49509218/pull/38/badge" alt="ReviewNinja"/>

meech-ward committed:
Add a note about not using @max.intValue

Add a note to the Largest Number assignment that tells the student not to use kvc mac.intValue to find the largest number in the array.

meech-ward committed:
Switch assignment 3 and 4

The largest number assignment should come first, after the collections challenge. The Toyotas assignment should come later, after the objects and inheritance challenges.

meech-ward committed:
Remove references to code school

Code school discontinued there objective-c course so we are removing all references to this course.

meech-ward committed:
Merge branch 'feature/remove-code-school' into phase-1

philosopherdog committed:
remove reference to code school resource in Introduction to Objective-C

philosopherdog committed:
fix spelling error

philosopherdog committed:
add paragraph break

philosopherdog committed:
Refactor enum question
- Renames enum type to follow convention of capping and adding type.
- Add switch part of the switch example.

philosopherdog committed:
Refactor Intro to Objects
- Clarify that NSString is immutable and all manipulations return a new instance.
- Edit

philosopherdog committed:
Clarify Structs, Stack, Heap

- Removed mention of high level overview which should point to a gist not a github issue discussion.
- Move chapters to read to top.
- Fix chapter numbers.
- Fix references to discussion of stack, heap, structs.

philosopherdog committed:
Refactor NSNumber, NSDate, NSDateFormatter

- Change name of Xcode project to reflect this section
- Add information on NSNumber as a wrapper of a primitive value.
- Add additional task to increment an NSNumber by unwrapping it.
- Reword format string section.
- Remove actual code for the formmater task!
- States that TimeInterval is a typedef.
- In Considerations section include mention of NSDate and the Formatter.

philosopherdog committed:
Refactor Collections

- Fix chapter line numbering for BNR
- Add note on creating NSArray using literal syntax, including code snippet and reference to brief blog post.
- Fix claim that you cannot change the contents of an immutable array to state that you cannot add or remove elements instead.
- Add more detailed summary

philosopherdog committed:
Fix Formatting On Assignment 3

- Use > to indent the note.

philosopherdog committed:
Edit Create Custom Classes

- Add BNR resource.
- Change word "object" to "instance" since "object" is ambiguous between class and instance.
- Add question to instruction that overwrites the property on person.

philosopherdog committed:
Edit Custom Methods Section

- Remove ambiguous use of "object"
- Add instructions for creating instances and logging out the results.

philosopherdog committed:
Edit Instrance Variables Section

- Replace "class" with the ambiguous "object".
- Change Chapter number to correspond to recent BNR version.
- Add additional video that explains manual properties.
- Add additional exercise to convert manual property to standard one.

philosopherdog committed:
Edit Properties Section

- Minor editing
- Suggestion: Move this section before the previous one on hand rolling your own properties.

philosopherdog committed:
Edit Initializers

- Add internal link to Creating Custom Class.
- Rewrite blurb on readonly properties. It's actually not true that readonly properties cannot be written to externally (KVC?). So, re-wrote this to make the statements true without going into complexities.
- Fill in missing summary.

philosopherdog committed:
Edit Inheritance Section

- Explain where "description" comes from and that it is called by NSLog.
- Explain that a superclass method without an implementation is usually called an "abstract method"
- Add Considerations that discuss the advantage of subclassing.

philosopherdog committed:
Edit Assignment 4

- Fix formatting.

philosopherdog committed:
Edit Congratulations Section

- Do minor edits.

philosopherdog committed:
Add Missing Link to NSDate Docs

- Add link missing in considerates section of More Objects

meech-ward committed:
Merge pull request #40 from lighthouse-labs/hotfix/phase-1

Hotfix/phase 1

meech-ward commented:
Moved to https://github.com/lighthouse-labs/iOS-Curriculum
